docker_custodian/docker_gc.py

```diff
@@ -21,6 +22,17 @@
 YEAR_ZERO = "0001-01-01T00:00:00Z"
 
 
+def list_events_since(client, since):
+    """
+    Retrieves a list of events that have happened "since"
+    the datetime provided or that happend in the last x seconds.
+    :param client: docker client
+    :param since: timedelta
+    """
+    events = client.events(since=dt.utcnow() - since, until=1, decode=True)
+    return [event for event in events]
```

Comment:
Maybe you could do something like have a long running process tail the events stream and track when images are used, and write out a relevant exclude list every few minutes?

It's more moving pieces, but it might be more accurate.